src/main/java/net/glease/ggfab/mte/MTE_LinkedInputBus.java

```diff
@@ -341,6 +345,64 @@ public void onScrewdriverRightClick(byte aSide, EntityPlayer aPlayer, float aX,
+            aPlayer.addChatMessage(new ChatComponentTranslation("ggfab.info.linked_input_bus.no_data"));
+            return true;
+        }
+        ItemStack circuit = GT_Utility.loadItem(stick.stackTagCompound, "circuit");
+        String channel = stick.stackTagCompound.getString("channel");
+        if (GT_Utility.isStackInvalid(circuit)) circuit = null;
+        if ("".equals(channel)) {
+            aPlayer.addChatMessage(new ChatComponentTranslation("ggfab.info.linked_input_bus.no_data"));
+            return true;
+        } else if (circuit != null && !getConfigurationCircuits().contains(circuit)) {
+            aPlayer.addChatMessage(new ChatComponentTranslation("ggfab.info.linked_input_bus.no_data"));
+            return true;
+        }
+        UUID owner = stick.stackTagCompound.hasKey("owner1") ?
+                new UUID(stick.stackTagCompound.getLong("owner1"), stick.stackTagCompound.getLong("owner2")) :
+                null;
+        if (owner != null && !owner.equals(getBaseMetaTileEntity().getOwnerUuid())) {
+            aPlayer.addChatMessage(new ChatComponentTranslation("ggfab.info.linked_input_bus.not_owned"));
+            return true;
+        }
+        setPrivate(owner != null);
+        setChannel(channel);
+        setInventorySlotContents(getCircuitSlot(), circuit);
+        aPlayer.addChatMessage(new ChatComponentTranslation("ggfab.info.linked_input_bus.data_pasted", channel));
+        return true;
+    }
+
+    @Override
+    public void onLeftclick(IGregTechTileEntity aBaseMetaTileEntity, EntityPlayer aPlayer) {
+        if (!(aPlayer instanceof EntityPlayerMP)) return;
+        ItemStack stick = aPlayer.inventory.getCurrentItem();
+        if (!ItemList.Tool_DataStick.isStackEqual(stick, true, true)) return;
+        if (getChannel() == null) {
+            aPlayer.addChatMessage(new ChatComponentTranslation("ggfab.info.linked_input_bus.no_channel"));
+            return;
+        }
+        NBTTagCompound tag = new NBTTagCompound();
+        tag.setString("ggfab.type", "linkedinputbus");
+        tag.setString("channel", getChannel());
+        tag.setTag("circuit", GT_Utility.saveItem(getStackInSlot(getCircuitSlot())));
+        if (isPrivate()) {
+            tag.setLong("owner1", getBaseMetaTileEntity().getOwnerUuid().getMostSignificantBits());
+            tag.setLong("owner2", getBaseMetaTileEntity().getOwnerUuid().getLeastSignificantBits());
+        }
+        aPlayer.addChatMessage(new ChatComponentTranslation("ggfab.info.linked_input_bus.data_copied", getChannel()));
+        stick.stackTagCompound = tag;
+        stick.setStackDisplayName("Linked Input Bus configuration");
+        // abuse the title mechanism here. I assure you it will be fine (tm).
+        GT_Utility.ItemNBT.setBookTitle(stick, "Channel: " + getChannel());
+        GT_Utility.ItemNBT.setBookAuthor(stick, getBaseMetaTileEntity().getOwnerName() == null ? "Mysteriously Unknown Person" : getBaseMetaTileEntity().getOwnerName());
```

Comment:
There are too many mysterious men...

```suggestion
        String ownerName = getBaseMetaTileEntity().getOwnerName();
        if (ownerName != null) {
            GT_Utility.ItemNBT.setBookAuthor(stick, ownerName);
        }
```